src/values.ts

```diff
@@ -1,21 +1,194 @@
+ * A wrapper aroud the Fauna `Date` type. It represents a calendar date that is
+ * not associated with a particular time or time zone, e.g. August 24th, 2006.
  *
- * @remarks This class follows the TC39 proposal for `Temporal.PlainDate`. See the
- * proposal here: https://tc39.es/proposal-temporal/docs/index.html.
- * Since fauna has both `Time` and `Date` data types, we want to make a clear
- * distinction between them and avoid using the native `Date` class.
+ * @remarks The Javascript `Date` type always has a time associated with it, but
+ * Fauna's `Date` type does not. When converting from a Fauna `Date` to a
+ * Javascript `Date`, we set time to 00:00:00 UTC. When converting a Javascript
+ * `Date` or time string to Fauna `Date`, we convert to UTC first. Care should
+ * be taken to specify the desired date, since Javascript `Date`s use local
+ * timezone info by default.
+ *
+ * TODO: link to beta docs for `Date` type
  */
-class FaunaDate {}
+export class DateStub {
+  readonly #dateString: string;
+
+  /**
+   * @remarks constructor is private to enforce using factory functions
+   */
+  private constructor(dateString: string) {
+    this.#dateString = dateString;
+  }
+
+  /**
+   * Creates a new {@link DateStub} from a date string
+   * @param dateString - A plain date string. The time is converted to UTC
+   * before saving the date.
+   * @returns A new {@link DateStub}
+   * @throws TypeError if a string is not provided, or RangeError if dateString
+   * is not a valid date
+   */
+  static from(dateString: string): DateStub {
+    if (typeof dateString !== "string") {
+      throw new TypeError(
+        `Expected string but received ${typeof dateString}: ${dateString}`
+      );
+    }
+    const matches = PARSE.plaindate.exec(dateString);
+    if (matches === null) {
+      throw new RangeError(
+        `Expected a plain date string but received '${dateString}'`
+      );
+    }
+    // There are some dates that match the regex but are invalid, such as Feb 31.
+    // Javascript does not parse all years that are valid in fauna, so let
+    // Fauna be the final check.
+
+    return new DateStub(matches[0]);
+  }
+
+  /**
+   * Creates a new {@link DateStub} from a Javascript `Date`
+   * @param date - A Javascript `Date`. The time is converted to UTC before
+   * saving the date.
+   * @returns A new {@link DateStub}
+   */
+  static fromDate(date: Date): DateStub {
+    if (!(date instanceof Date)) {
+      throw new TypeError(`Expected Date but received ${typeof date}: ${date}`);
+    }
+    const dateString = date.toISOString();
+    const matches = PARSE.startsWithPlaindate.exec(dateString);
+    if (matches === null) {
+      // Our regex should match any possible date that comes out of
+      // `Date.toISOString()`, so we will only get here if the regex is
+      // incorrect. This is a ClientError since it is our fault.
+      throw new ClientError(`Failed to parse date '${date}'`, { cause: null });
```

Comment:
```suggestion
      throw new ClientError(`Failed to parse date '${date}'`);
```

Reply:
Removing this causes a type error. We can make the `options` parameter options when constructing `ClientError`.

Reply:
ah okay yeah seems like it should be optional